openedx/core/djangoapps/appsembler/settings/settings/common.py

```diff
@@ -37,3 +37,5 @@ def plugin_settings(settings):
 
     settings.CUSTOM_DOMAINS_REDIRECT_CACHE_TIMEOUT = None  # The length of time we cache Redirect model data
     settings.CUSTOM_DOMAINS_REDIRECT_CACHE_KEY_PREFIX = 'custom_domains_redirects'
+
+    settings.COPY_SEGMENT_EVENT_PROPERTIES_TO_TOP_LEVEL = True
```

Comment:
```suggestion
    settings.COPY_SEGMENT_EVENT_PROPERTIES_TO_TOP_LEVEL = False
```
I think it should be disabled by default, so specific customers that needs it can turn it on.

Reply:
I don't know...I can understand wanting to be conservative with a new feature but  I disagree.  The majority of Segment Destinations won't be able to use TRACK events without this so the typical case will be broken.  The only way some TRACK events work now is because in edx-platform, edX themselves have just bypassed event-tracking altogether, and just directly called `analytics.track()` (Segment's Python API method) from either Python code or JS code.  That's a bad pattern and I will try to encourage them to remove all of those.  

I think if we leave this defaulted to off, customers who do want to make use of a wider array of TRACK events will always have to make a change to get that to work, and we would more likely be left support two event structures.  I think we should let the few customers who already have integrations (they probably aren't relying much on TRACK events) without it have it turned off but have everybody move to the new format which should work for just about any use case.

Reply:
The other thing I considered is making it configurable to just remove the properties from the `data` object that have been copied up to the top level.  Probably that should be the default, too.

Reply:
@bryanlandia I understand the point and I know it's a simple modification, but at least, we need to coordinate with CS team and let the customers know before rolling the change. So we can merge this PR with this default to False, and after we inform customers, we turn it as a default if they're ok, but we cannot ship it without notice.